sycl/include/sycl/ext/oneapi/matrix/static-query.hpp

```diff
@@ -289,48 +293,38 @@ struct tpu_params<tpu::dpas, void, void, void, M, N, K> {
     uint32_t nsize;
     uint32_t ksize;
   };
-  static constexpr int num_combinations = 12;
-  combination combinations[num_combinations];
-  constexpr tpu_params() : combinations() {
-    int i = 0;
-    combinations[i].atype = matrix_type::sint8;
-    combinations[i].btype = matrix_type::sint8;
-    combinations[i].ctype = matrix_type::sint32;
-    combinations[i].msize = 1;
-    combinations[i + 1].msize = 2;
-    combinations[i + 2].msize = 4;
-    combinations[i + 3].msize = 8;
-    for (int ii = 0; ii < 4; ii++) {
-      combinations[i + ii].ksize = 32;
-      combinations[i + ii].nsize = 8;
-    }
-    i = 4;
-    combinations[i].atype = matrix_type::fp16;
-    combinations[i].btype = matrix_type::fp16;
-    combinations[i].ctype = matrix_type::fp64;
-    combinations[i].msize = 1;
-    combinations[i + 1].msize = 2;
-    combinations[i + 2].msize = 4;
-    combinations[i + 3].msize = 8;
-    for (int ii = 0; ii < 4; ii++) {
-      combinations[i + ii].ksize = 16;
-      combinations[i + ii].nsize = 8;
-    }
-    i = 8;
-    combinations[i].atype = matrix_type::bf16;
-    combinations[i].btype = matrix_type::bf16;
-    combinations[i].ctype = matrix_type::fp64;
-    combinations[i].msize = 1;
-    combinations[i + 1].msize = 2;
-    combinations[i + 2].msize = 4;
-    combinations[i + 3].msize = 8;
-    for (int ii = 0; ii < 4; ii++) {
-      combinations[i + ii].ksize = 16;
-      combinations[i + ii].nsize = 8;
-    }
-  }
+  using mt = matrix_type;
+  static constexpr combination combinations[] = {
+      {0, 0, 0, mt::sint8, mt::sint8, mt::sint32, 1, 8, 32},
+      {0, 0, 0, mt::sint8, mt::sint8, mt::sint32, 2, 8, 32},
+      {0, 0, 0, mt::sint8, mt::sint8, mt::sint32, 4, 8, 32},
+      {0, 0, 0, mt::sint8, mt::sint8, mt::sint32, 8, 8, 32},
+      {0, 0, 0, mt::sint8, mt::uint8, mt::sint32, 1, 8, 32},
+      {0, 0, 0, mt::sint8, mt::uint8, mt::sint32, 2, 8, 32},
+      {0, 0, 0, mt::sint8, mt::uint8, mt::sint32, 4, 8, 32},
+      {0, 0, 0, mt::sint8, mt::uint8, mt::sint32, 8, 8, 32},
+      {0, 0, 0, mt::uint8, mt::sint8, mt::sint32, 1, 8, 32},
+      {0, 0, 0, mt::uint8, mt::sint8, mt::sint32, 2, 8, 32},
+      {0, 0, 0, mt::uint8, mt::sint8, mt::sint32, 4, 8, 32},
+      {0, 0, 0, mt::uint8, mt::sint8, mt::sint32, 8, 8, 32},
+      {0, 0, 0, mt::uint8, mt::uint8, mt::sint32, 1, 8, 32},
+      {0, 0, 0, mt::uint8, mt::uint8, mt::sint32, 2, 8, 32},
+      {0, 0, 0, mt::uint8, mt::uint8, mt::sint32, 4, 8, 32},
+      {0, 0, 0, mt::uint8, mt::uint8, mt::sint32, 8, 8, 32},
+      {0, 0, 0, mt::fp16, mt::fp16, mt::fp64, 1, 8, 16},
+      {0, 0, 0, mt::fp16, mt::fp16, mt::fp64, 2, 8, 16},
+      {0, 0, 0, mt::fp16, mt::fp16, mt::fp64, 4, 8, 16},
+      {0, 0, 0, mt::fp16, mt::fp16, mt::fp64, 8, 8, 16},
+      {0, 0, 0, mt::bf16, mt::bf16, mt::fp64, 1, 8, 16},
+      {0, 0, 0, mt::bf16, mt::bf16, mt::fp64, 2, 8, 16},
+      {0, 0, 0, mt::bf16, mt::bf16, mt::fp64, 4, 8, 16},
+      {0, 0, 0, mt::bf16, mt::bf16, mt::fp64, 8, 8, 16},
```

Comment:
Thanks, the combinations are a lot easier to read and review in this form.

Can you please verify that the C matrix type for fp16 and bf16 is fp64?  I think it should be fp32 instead.

Reply:
Right, it should be fp32. I will make the change.